src/hera/workflow.py

```diff
@@ -110,6 +113,8 @@ def __init__(
         metrics: Optional[Union[Metric, List[Metric], Metrics]] = None,
     ):
         self.name = validate_name(name)
+        dag_name = dag_name if dag_name is not None else self.name
```

Comment:
This might be easier to understand
```suggestion
        dag_name = self.name is dag_name is None else dag_name
```